scene/gui/scroll_container.cpp

```diff
@@ -288,15 +288,15 @@ void ScrollContainer::_update_dimensions() {
 		child_max_size.y = MAX(child_max_size.y, minsize.y);
 
 		Rect2 r = Rect2(-Size2(get_h_scroll(), get_v_scroll()), minsize);
-		if (horizontal_scroll_mode == SCROLL_MODE_DISABLED || (!h_scroll->is_visible_in_tree() && c->get_h_size_flags() & SIZE_EXPAND)) {
+		if (horizontal_scroll_mode == SCROLL_MODE_DISABLED || !h_scroll->is_visible_in_tree() || !need_h_scroll) {
 			r.position.x = 0;
 			if (c->get_h_size_flags() & SIZE_EXPAND) {
 				r.size.width = MAX(size.width, minsize.width);
 			} else {
 				r.size.width = minsize.width;
 			}
 		}
-		if (vertical_scroll_mode == SCROLL_MODE_DISABLED || (!v_scroll->is_visible_in_tree() && c->get_v_size_flags() & SIZE_EXPAND)) {
```

Comment:
I don't know why this condition would check size flags previously, but it seems to have no visible effect.